### PR TITLE
uef commander projectile fix a la jip

### PR DESCRIPTION
**Pull request by brokerjoe**
This branch is for bugfixes, general improvements, and new features. If your change is designed to alter balance, please make 
sure your changes are rebased onto [development/balance] (https://github.com/FAForever/fa/tree/development/balance), and target that branch with your PR.

### CODE REVIEW

**Comment by brokerjoe** [comment]
Watched, the video and thought I setup real quick the development environment and try it out myself, happy with the progress, so I thought I submit the adoption fix. The overall performance impact will be not as big as the engineer class, but following the same pattern is important. Its the sum of all.

**Comment by Garanas** [comment]
Jip here: looking good 😄

File: lua/EffectUtilitiesUEF.lua [hunk 1]
```diff
@@ -371,79 +371,169 @@ end
...
-            beamEndBuilder:SetVelocity(velX, velY, velZ)
-            Warp(beamEndBuilder2, Vector(cx2, (cy2 - (oy * unitBeingBuilt:GetFractionComplete())), cz2))
-            beamEndBuilder2:SetVelocity(-velX, -velY, -velZ)
+            vc[1] = fx1
+            vc[2] = (fy - (oy * fraction))
+            vc[3] = fz1
+            Warp(beamEndBuilder, vc)
+            ProjectileSetVelocity(beamEndBuilder, velX, 0, velZ)
+            vc[1] = fx2
+            vc[2] = (fy - (oy * UnitGetFractionComplete(unitBeingBuilt)))
```
Comment (Garanas) [inline lua/EffectUtilitiesUEF.lua:519]: This is the same as `fraction`, on line 510

File: lua/EffectUtilitiesUEF.lua [hunk 2]
```diff
@@ -371,79 +371,169 @@ end
...
+            Warp(beamEndBuilder2, vc)
+            ProjectileSetVelocity(beamEndBuilder2, -velX, 0, -velZ)
             flipDirection = false
         else
-            Warp(beamEndBuilder, Vector(cx2, (cy2 - (oy * unitBeingBuilt:GetFractionComplete())), cz2))
-            beamEndBuilder:SetVelocity(-velX, -velY, -velZ)
-            Warp(beamEndBuilder2, Vector(cx1, (cy1 - (oy * unitBeingBuilt:GetFractionComplete())), cz1))
-            beamEndBuilder2:SetVelocity(velX, velY, velZ)
+            vc[1] = fx2
+            vc[2] = (fy - (oy * UnitGetFractionComplete(unitBeingBuilt)))
```
Comment (Garanas) [inline lua/EffectUtilitiesUEF.lua:526]: This is the same as `fraction`, on line 510

**Comment by Garanas** [comment]
I'll review this tomorrow again and merge it accordingly. Great work!

**Comment by Garanas** [comment]
Do you have suggestions on a global linter? I have not looked at it myself up to this point

**Comment by brokerjoe** [comment]
See https://github.com/FAForever/fa/pull/3745 ; it matters to the automation more then it does in our case. Here we have a lot of review processes and the style can easily enforced by the reviewer, also its spare time programming who wants linting there.

**Comment by Garanas** [comment]
@brokerjoe What name would you like to show up in the changelog? Typically people use their in-game name, but feel free to use any other name.

**Comment by brokerjoe** [comment]
brokerjoe is fine

**Pull request merged**